File: test/resources/newSyntax/withTypeAnnotations/otherExamples/rev_acc/rev_acc/rev_acc/cpd.pl
rev(cons(Q1, Q2)) :- _rev(Q2, cons(Q1, nil)).
_rev(nil, cons(a, cons(b, cons(c, cons(d, nil))))).
_rev(cons(Q1, Q2), Y2) :- _rev(Q2, cons(Q1, Y2)).

File: test/resources/newSyntax/withTypeAnnotations/otherExamples/rev_acc/rev_acc/rev_acc/cpd.ml
open GT
open OCanren
open OCanren.Std
open Helper

let topLevel x0 = 
  let rec rev y0 = (fresh (q1 q2) (((y0 === (q1 % q2)) &&& (_rev q2 ((q1 % ((List.nil ())))))))) 
  and _rev y1 y2 = (fresh (q1 q2) ((((y2 === (((a ())) % ((((b ())) % ((((c ())) % ((((d ())) % ((List.nil ())))))))))) &&& (y1 === (List.nil ()))) ||| ((y1 === (q1 % q2)) &&& (_rev q2 ((q1 % y2))))))) 
  in   (rev x0)
